src/main/kotlin/com/katanox/tabour/core/ProtoEventConsumer.kt

```diff
@@ -0,0 +1,12 @@
+package com.katanox.tabour.core
+
+import com.google.protobuf.Message
+import com.google.protobuf.MessageOrBuilder
+import com.google.protobuf.util.JsonFormat
+
+abstract class ProtoEventConsumer<T: Message.Builder>: EventConsumer() {
```

Comment:
no really doable here, as this function will be overridden when you consume therefore the returned value can stay as a byte array or a string.